### PR TITLE
Slightly better miniselect usage

### PR DESCRIPTION
**Pull request by abyss7**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Review by alexey-milovidov — APPROVED** [review]

File: src/Columns/ColumnDecimal.cpp [hunk 1]
```diff
@@ -197,21 +195,11 @@ void ColumnDecimal<T>::updatePermutation(bool reverse, size_t limit, int, IColum
...
+            partial_sort(res.begin() + first, res.begin() + limit, res.begin() + last,
                 [this](size_t a, size_t b) { return data[a] > data[b]; });
-#endif
         else
-#if !defined(ARCADIA_BUILD)
-            miniselect::floyd_rivest_partial_sort(res.begin() + first, res.begin() + limit, res.begin() + last,
-                [this](size_t a, size_t b) { return data[a] < data[b]; });
-#else
-            std::partial_sort(res.begin() + first, res.begin() + limit, res.begin() + last,
+            partial_sort(res.begin() + first, res.begin() + limit, res.begin() + last,
```
Comment (alexey-milovidov) [inline src/Columns/ColumnDecimal.cpp:201]: PVS-Studio argues for reason.

PS. Let's check if tests will pass, and if they will do - we need to add a new test.

**Review by alexey-milovidov — CHANGES_REQUESTED** [review]
Looks like tests don't cover updatePermutation for ORDER BY Decimal field DESC.

File: src/Columns/ColumnDecimal.cpp (hunk 1 above)
Comment (abyss7) [reply src/Columns/ColumnDecimal.cpp:201]: I see. The problem was only in the Arcadia build before.

**Review by alexey-milovidov — APPROVED** [review]

**Comment by abyss7** [comment]
Yandex check failure looks unrelated

**Pull request merged**